libs/table/src/lib/directives/tree-grid/tree-grid.directive.ts

```diff
@@ -0,0 +1,102 @@
+import { Directive, HostBinding, Input } from '@angular/core';
+
+import { NgxTreeGridRowDirective } from './tree-grid-row.directive';
+import { NgxTreeGridCellDirective } from './tree-grid.cell.directive';
+
+/**
+ * An overarching directive to handle navigation according to the WCAG treegrid pattern
+ *
+ * See https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/
+ */
+@Directive({
+	selector: '[ngxTreeGrid]',
+	standalone: true,
+})
+export class NgxTreeGridDirective {
+	/**
+	 * An array of all rows in the treegrid
+	 */
+	private rows: NgxTreeGridRowDirective[] = [];
+
+	@HostBinding('attr.role') private role: 'table' | 'treegrid' = 'table';
```

Comment:
Maybe put this union type in the tree-grid types, for convenience?

Reply:
Same as above, there's no use for this type anywhere else, so I prefer to keep it here.